gokart/task.py

```diff
@@ -373,6 +374,18 @@ def make_unique_id(self) -> str:
             self.task_unique_id = unique_id
         return unique_id
 
+    def to_str_params(self, only_significant=False, only_public=False):
+        if only_significant and (not only_public):
+            # cache to_str_params to avoid too slow task creation of deep task tree
+            # e.g. gokart.build(RecursiveTask(dep=RecursiveTask(dep=RecursiveTask(dep=HelloWorldTask())))) takes O(n*n) to_str_params calls
```

Comment:
[nits]
This comment make sense to me but little ambiguous, e.g. there's no definition of `n`.
To make this clear, how about followings:
```suggestion
            # e.g. gokart.build(RecursiveTask(dep=RecursiveTask(dep=RecursiveTask(dep=HelloWorldTask())))) needs O(n^2) times to_str_params calls with respect to n times RecursiveTask
```

Reply:
@hirosassa THX!!